src/util/parented_ptr.h

```diff
@@ -104,6 +88,36 @@ inline parented_ptr<T> make_parented(Args&&... args) {
     return parented_ptr<T>(new T(std::forward<Args>(args)...));
 }
 
+template<typename T, typename U>
+inline bool operator== (const T* lhs, const parented_ptr<U>& rhs) {
+    return lhs == rhs.get();
+}
+
+template<typename T, typename U>
+inline bool operator== (const parented_ptr<T>& lhs, const T* rhs) {
+    return lhs.get() == rhs;
+}
+
+template<typename T>
+inline bool operator== (const parented_ptr<T>& lhs, const parented_ptr<T>& rhs) const {
+    return lhs.get() == rhs.get();
+}
+
+template<typename T, typename U>
+inline bool operator!= (const T* lhs, const parented_ptr<U>& rhs) {
+    return lhs != rhs.get();
+}
+
+template<typename T, typename U>
+inline bool operator!= (const parented_ptr<T>& lhs, const T* rhs) {
+    return lhs.get() != rhs;
+}
+
+template<typename T>
```

Comment:
Would be more versatile by adding a second generic type U for the right-hand side